src/tools/debugger/dosdebug.c

```diff
@@ -172,8 +174,14 @@ static void handle_console_input(void)
         kill_timeout=KILL_TIMEOUT;
       }
       write(fdout, buf, n);
-      memcpy(sbuf, buf, n);
-      sn=n;
+
+      if (strncmp(buf, "d ", 2) == 0)
+        sn = snprintf(sbuf, sizeof sbuf, "d\n");
```

Comment:
strcpy()? :)

Reply:
With snprintf I can set the sn var at the same time as well as avoiding the buffer overrun. It's a habit of mine. I can change it if it offends?

Reply:
OK, I didn't realize `sn` is used later, it is
invisible in the patch context.